src/inventory/inventory.py

```diff
@@ -19,33 +21,89 @@ def __init__ (self):
 
-    # TODO, reimagine each of these find_cards_* methods, as filters methods.
-    # Provide them with the list that is to be filtered.
-    # This will allow complex chaining, of searches.
-    def find_cards_with_set(self,searched_setname):
-        '''Returns a list of Dataframes for each card in the ManaPool-Inventory.db matching the provided setname'''
-        return f"Stub, but here's {searched_setname}"
-        #return self.cursor.execute(f"SELECT * FROM 'MTG-Cards' WHERE set_name={searched_setname}'")
-
-    def find_cards_with_name(self, searched_cardname):
-        '''Returns a list of Dataframes for each card in the ManaPool-Inventory.db matching the provided card name'''
-        return f"Stub, but here's {searched_cardname}"
-        #return self.cursor.execute(f"SELECT * FROM 'MTG-Cards' WHERE name={searched_cardname}'")
-
+   # TODO: Add count values to database to track cards contained in inventory
+    def searchBySet(self, setname):
+        query = f"SELECT id, multiverse_ids, name, mana_cost, type_line, set_name, rarity FROM '{self.table_name}' WHERE set_name='{setname}'"
+        res = self.cursor.execute(query)
+        return self.__dbResponseHandler(res)
+
+   # TODO: Add count values to database to track cards contained in inventory
+    def searchByBlock(self, blockname):
+        query = f"SELECT id, multiverse_ids, name, mana_cost, type_line, set_name, rarity FROM '{self.table_name}' WHERE block='{blockname}'"
+        res = self.cursor.execute(query)
+        return self.__dbResponseHandler(res)
+
+   # TODO: Add count values to database to track cards contained in inventory
+    def searchByName(self, cardname):
+        query = f"SELECT id, multiverse_ids, name, mana_cost, type_line, set_name, rarity FROM '{self.table_name}' WHERE name='{cardname}'"
+        # test_multi = f"SELECT multiverse_ids, name, mana_cost, type_line, set_name, rarity  FROM 'MTG-Cards' WHERE name='' or name=''"
+        # cards = pd.to_sql(query, self.connection)
+        # self.cursor.execute(f"SELECT * FROM 'MTG-Cards' WHERE set_name={cardname}'")
+        res = self.cursor.execute(query)
+        return self.__dbResponseHandler(res)
+    
+   # TODO: Add count values to database to track cards contained in inventory
+    def searchByMID(self, card_mid):
+        query = f"SELECT id, multiverse_ids, name, mana_cost, type_line, set_name, rarity FROM '{self.table_name}' WHERE multiverse_ids='{card_mid}'"
+        # test_multi = f"SELECT multiverse_ids, name, mana_cost, type_line, set_name, rarity  FROM 'MTG-Cards' WHERE multiverse_ids='109722' OR multiverse_ids='189637'"
+        res = self.cursor.execute(query)
+        return self.__dbResponseHandler(res)
+
+   # TODO: Add count values to database to track cards contained in inventory
+   # DB queries return with an iterator of tuples.
+   # __dbResponseHandler parses these tuples into a dict that can be consumed by the client.
+    def __dbResponseHandler(self, res_iter):
```

Comment:
A `Connection` object from SQLite uses a function, `row_factory` that generates each resulting row of the SQL queries through the connection.
 Define a function `dict_factory` that overrides how SQLite generates rows, so that it generates dictionaries instead of tuples, then set the row `row_factory` function to your new `dict_factory` function.

See the SQLite documentation for details and an example: 
https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.row_factory